lib/Dialect/TritonNvidiaGPU/IR/Ops.cpp

```diff
@@ -261,10 +262,79 @@ static void printToken(OpAsmPrinter &p, Operation *op, Value dep, Type token) {
   p << ']';
 }
 
+namespace {
+enum class MMADTypeKind { tf32, f16, f8f6f4, i8 };
+} // namespace
+
+static std::string strMMADTypeKind(MMADTypeKind kind) {
+  switch (kind) {
+  case MMADTypeKind::tf32:
+    return "tf32";
+  case MMADTypeKind::f16:
+    return "f16";
+  case MMADTypeKind::f8f6f4:
+    return "f8f6f4";
+  case MMADTypeKind::i8:
+    return "i8";
+  }
+}
+
 LogicalResult TCGen5MMAOp::verify() {
+  auto getDTypeKindAndAcc =
+      [](Type t) -> std::optional<std::pair<MMADTypeKind, SmallVector<Type>>> {
+    MLIRContext *ctx = t.getContext();
+    if (t.isF32()) {
+      return {{MMADTypeKind::tf32, {Float32Type::get(ctx)}}};
+    }
+    if (t.isF16() || t.isBF16()) {
+      return {
+          {MMADTypeKind::f16, {Float16Type::get(ctx), Float32Type::get(ctx)}}};
+    }
+    if (isa<Float8E4M3FNType, Float8E5M2Type, Float6E2M3FNType,
+            Float6E3M2FNType, Float4E2M1FNType>(t)) {
```

Comment:
not all those types are supported right? Also we currently use i8 for sub-byte types. Wouldn't that be a problem?